contrib/pg_tde/t/pgtde.pm

```diff
@@ -81,4 +81,65 @@ sub compare_results
 	return compare($expected_filename_with_path, $out_filename_with_path);
 }
 
+# Common TDE helpers
+
+# Check if the encryption status of a table is as expected and return 't' or 'f'
+sub check_encryption_status
+{
+	my ($node, $table_name, $expected) = @_;
+	my $result =
+	  safe_psql('postgres', "SELECT pg_tde_is_encrypted('$table_name')");
+	append_to_result_file($node->name . ": encryption check result for $table_name = $result");
+	is($result, $expected, "Check encryption status for '$table_name' on " . $node->name);
+}
+
+# Set up pg_tde extension and add a global key provider and set the server key
+sub setup_pg_tde_global_environment
+{
+	my ($node, $key_name, $provider_name, $provider_path) = @_;
+	psql($node, 'postgres', 'CREATE EXTENSION IF NOT EXISTS pg_tde;');
+	psql($node, 'postgres',
+		"SELECT pg_tde_add_global_key_provider_file('$provider_name', '$provider_path');");
+	psql($node, 'postgres',
+		"SELECT pg_tde_set_server_key_using_global_key_provider('$key_name', '$provider_name');");
+}
+
+# Set up pg_tde extension and add a database key provider and set the database key
+sub setup_pg_tde_db_environment
+{
+	my ($node, $key_name, $provider_name, $provider_path) = @_;
+	psql($node, 'postgres', 'CREATE EXTENSION IF NOT EXISTS pg_tde;');
+	psql($node, 'postgres',
+		"SELECT pg_tde_add_database_key_provider_file('$provider_name', '$provider_path');");
+	psql($node, 'postgres',
+		"SELECT pg_tde_set_key_using_database_key_provider('$key_name', '$provider_name');");
+}
+
+# Set up pg_tde in postgresql.conf
+sub enable_pg_tde_in_conf
+{
+	my ($node) = @_;
+	$node->append_conf('postgresql.conf', "shared_preload_libraries = 'pg_tde'");
+}
+
+# Set default table access method to tde_heap
+sub set_default_table_am_tde_heap
+{
+	my ($node) = @_;
+	$node->append_conf('postgresql.conf', "default_table_access_method = 'tde_heap'");
+}
```

Comment:
I'm not sure about those two functions. I don't think anyone would memorize this func names, hence they would be copied from the existing tests anyway. But then it's not a bit difference which line is to copy (I mean `$node->append_conf('postgresql.conf', "default_table_access_method = 'tde_heap'");` or `set_default_table_am_tde_heap($node)`). But `append_conf(` is more clear of what it does. However, I don't have super strong opinion about that.